### PR TITLE
Preparing release 2.0.0

### PR DESCRIPTION
**Pull request by gvolpe**
Closes #234 and closes #235 

- Dropping support for Scala 2.11.x as Circe does not support it. 
- Updating and unifying dependencies for the different Scala versions. 
- Bumping version.

Pinging @Daenyth to get the green light on this one.

### CODE REVIEW

**Comment by gvolpe** [comment]
Leaving this here before I forget... It would be nice to get #230 done before a release. I think I can work on it this week.

**Comment by ybasket** [comment]
Nice - looking forward to this release!

Maybe upgrade to Scala 2.12.10 as well?

**Comment by gvolpe** [comment]
Yeah, I can do that too if there's no big issues! I didn't know 2.12.10 was out :smile:

File: build.sbt [hunk 1]
```diff
@@ -8,28 +8,20 @@ name := """fs2-rabbit-root"""
 organization in ThisBuild := "dev.profunktor"
 
 scalaVersion in ThisBuild := "2.12.9"
-crossScalaVersions in ThisBuild := Seq("2.11.12", "2.12.9", "2.13.0")
-
-// makes `tut` fail :( -> https://github.com/tpolecat/tut/issues/255
-//scalaVersion in ThisBuild := "2.12.8" // needed for metals
+crossScalaVersions in ThisBuild := Seq("2.12.9", "2.13.0")
```
Comment (Daenyth) [inline build.sbt]: Do we want to set this to 2.12.10?
```suggestion
crossScalaVersions in ThisBuild := Seq("2.12.10", "2.13.0")
```

File: build.sbt [hunk 2]
```diff
@@ -117,44 +91,51 @@ lazy val noPublish = Seq(
...
+lazy val `fs2-rabbit-circe` = project
+  .in(file("json-circe"))
   .settings(commonSettings: _*)
   .settings(libraryDependencies ++= JsonDependencies(scalaVersion.value))
   .settings(parallelExecution in Test := false)
   .enablePlugins(AutomateHeaderPlugin)
   .dependsOn(`fs2-rabbit`)
 
-lazy val `fs2-rabbit-test-support` = project.in(file("test-support"))
+lazy val `fs2-rabbit-test-support` = project
```
Comment (Daenyth) [inline build.sbt]: Is this published? I should check that out, might be handy in my project :)

File: project/Dependencies.scala [hunk 3]
```diff
@@ -2,12 +2,15 @@ import sbt._
...
+  object Version {
     val catsEffect = "2.0.0"
     val fs2        = "2.0.0"
-    val circe      = "0.11.1"
+    val circe      = "0.12.1"
     val amqpClient = "5.7.3"
     val logback    = "1.2.3"
+    val monix      = "3.0.0"
+    val zio        = "1.0.0-RC12-1"
+    val zioCats    = "2.0.0.0-RC3"
```
Comment (Daenyth) [inline project/Dependencies.scala]: I feel like we should only release ourselves as `RC` if one of our dependencies is at `RC`

File: version.sbt [hunk 4]
```diff
@@ -1,2 +1,2 @@
-version in ThisBuild := "2.0.0-RC3"
+version in ThisBuild := "2.0.0"
```
Comment (Daenyth) [inline version.sbt]:
```suggestion
version in ThisBuild := "2.0.0-RC4"
```

Because of the zio dependency only. Your call

File: build.sbt (hunk 1 above)
Comment (gvolpe) [reply build.sbt]: Yup, I just made the change in my local branch! :smile: I was making sure everything compiles before pushing.

File: build.sbt (hunk 2 above)
Comment (gvolpe) [reply build.sbt]: It surely is :) https://search.maven.org/search?q=a:fs2-rabbit-test-support_2.13

File: project/Dependencies.scala (hunk 3 above)
Comment (gvolpe) [reply project/Dependencies.scala]: These are only for the examples.

File: version.sbt (hunk 4 above)
Comment (gvolpe) [reply version.sbt]: `monix` and `zio` are only pulled by the `examples` which are not published as a dep.

**Pull request merged**